### PR TITLE
Changes in instances for upcoming feature selection framework

### PR DESCRIPTION
**Pull request by jpbarddal**

### CODE REVIEW

File: moa/src/main/java/com/yahoo/labs/samoa/instances/FilteredSparseInstance.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+ * language governing permissions and limitations under the
+ * License.  
+ */
+package com.yahoo.labs.samoa.instances;
+
+import moa.streams.ExampleStream;
+
+/**
+ * The Class FilteredSparseInstance.
+ *
```
Comment (abifet) [inline moa/src/main/java/com/yahoo/labs/samoa/instances/FilteredSparseInstance.java]: Please, can you add a short explanation of the motivation of this class and why it's needed?

File: moa/src/main/java/com/yahoo/labs/samoa/instances/FilteredSparseInstanceData.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the
+ * License.  
+ */
+package com.yahoo.labs.samoa.instances;
+
+/**
+ * The Class SparseInstanceData.
+ *
```
Comment (abifet) [inline moa/src/main/java/com/yahoo/labs/samoa/instances/FilteredSparseInstanceData.java]: Please, change the title, and add more details.

File: moa/src/main/java/com/yahoo/labs/samoa/instances/Instances.java [hunk 3]
```diff
@@ -61,6 +61,16 @@
...
     protected HashMap<String, Integer> hsAttributesIndices;
 
+    /**
+     * Indices of relevant features.
+     */
+    protected int[] indicesRelevants;
+
+    /**
+     * Indices of relevant features.
+     */
```
Comment (abifet) [inline moa/src/main/java/com/yahoo/labs/samoa/instances/Instances.java]: irrelevant features

**Comment by jpbarddal** [comment]
Changes made.

**Pull request merged**